### PR TITLE
improve_diff_handling: Update description

### PR DESCRIPTION
**Pull request by satwikkansal**
* Add more details
* Add sample mockups
* Add deliverables
* Update Milestones (more tasks)

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: _projects/improve_diff_handling.md [hunk 1]
```diff
@@ -41,11 +37,108 @@ corrections or gramatical corrections.
 In addition Bears will be able to offer templated patches, e.g. to offer
 variable renamings while the user should be asked for the new variable name.
 
+#### Sample mockups
+
+Source: https://docs.google.com/document/d/1BLZO2tD93yKdO7XhMb5NgJvytu-pWDtPMRidGmjQ8EY/edit?usp=sharing (You are free to use ideas discussed in the proposal).
+
+Candidates can take help of following sample mockups to understand the use cases better. The mockups discussed below however need not be strictly followed. Candidate is enocouraged than welcome to incorporate his ideas and designs.
```
Comment (jayvdb) [inline _projects/improve_diff_handling.md]: enocouraged - spelling... but, use stronger language, and gender neutral when possible:

> Candidate is expected to incorporate their own ideas and designs.

File: _projects/improve_diff_handling.md [hunk 2]
```diff
@@ -65,3 +158,4 @@ variable renamings while the user should be asked for the new variable name.
 * The feature of templated patches is merged and fully tested.
 * The new API is fully documented on <https://api.coala.io/>.
 * At least one Bear exists that makes use of this.
+* Output formats for the enhance features are added and their processing functions are implemented in the `Linter` class.
```
Comment (jayvdb) [inline _projects/improve_diff_handling.md]: grammar issue with `for the enhance features`  enhanced?

File: _projects/improve_diff_handling.md [hunk 3]
```diff
@@ -15,24 +15,20 @@ mentors:
...
-every issue in the code can have just a single correct solution. The fix can
-and should also be modifiable by the preferences of the user. Thus the
-following improvements can be incorporated giving coala the ability to:
-
-1. Offer multiple diffs, i.e. multiple different fixes to the same issue.
+coala provides static analysis routines in the form of bears that have capabilities to detect or even fix the problematic code. coala converts the fixes provided by different bears to its own `Diff` class instance which is later used to display the fix along with the analysis results to the users and let the users choose appropriate actions. This mechanism works smoothly as of now and even applies the proposed diff automatically to the problematic file if the user so desires. This is very helpful to the users. At the same time, not every issue in the code can have just a single correct solution. The fix can and should also be modifiable by the preferences of the user.
+
+The primary objective of the project is to improve the existing diff handling capability of coala so that it is possible for the bears to do things like:
```
Comment (jayvdb) [inline _projects/improve_diff_handling.md]: blank line after `:`

File: _projects/improve_diff_handling.md [hunk 4]
```diff
@@ -15,24 +15,20 @@ mentors:
...
-and even applies the proposed diff automatically to the problematic file if the
-user so desires. This is very helpful to the users. At the same time, not
-every issue in the code can have just a single correct solution. The fix can
-and should also be modifiable by the preferences of the user. Thus the
-following improvements can be incorporated giving coala the ability to:
-
-1. Offer multiple diffs, i.e. multiple different fixes to the same issue.
+coala provides static analysis routines in the form of bears that have capabilities to detect or even fix the problematic code. coala converts the fixes provided by different bears to its own `Diff` class instance which is later used to display the fix along with the analysis results to the users and let the users choose appropriate actions. This mechanism works smoothly as of now and even applies the proposed diff automatically to the problematic file if the user so desires. This is very helpful to the users. At the same time, not every issue in the code can have just a single correct solution. The fix can and should also be modifiable by the preferences of the user.
```
Comment (jayvdb) [inline _projects/improve_diff_handling.md]: split lines at 80cpl

**Comment by satwikkansal** [comment]
@jayvdb ping!

**Comment by jayvdb** [comment]
ack https://github.com/coala/projects/pull/494/commits/8615e41afa702fea2ef6802bdf1840b330c5da8f https://github.com/coala/projects/pull/494/commits/aa728fe60867f77094ba1e99d43c70c0ea091d59

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
reack https://github.com/coala/projects/pull/494/commits/1b10c49d082b2e4249fa28c17791346906c557fd https://github.com/coala/projects/pull/494/commits/90e3b788cbe3745597219ec4534323fe90cb350b

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**